tests/ui/offset-of/offset-of-unsized.rs

```diff
@@ -10,6 +9,8 @@ struct S<T: ?Sized> {
 trait Tr {}
 
 fn main() {
-    let _a = core::mem::offset_of!(S<dyn Tr>, a);
-    let _b = core::mem::offset_of!((u64, dyn Tr), 0);
+    let _a = core::mem::offset_of!(S<dyn Tr>, a); // ok
```

Comment:
I think it should be better to edit `offset-of-dst-field.rs` instead, or create a completely new file from a copy of this file, because right now the test would probably not trigger the ICE from #112051 (not 100% sure but IIRC codegen is not ran when type checking fails) ...

Reply:
Oops, thank you. Added a comment noting this too